src/test/java/com/wikia/webdriver/common/logging/Log.java

```diff
@@ -0,0 +1,338 @@
+
+  public static void image(String command, File image, boolean success) {
+    byte[] bytes = new byte[0];
+    try {
+      bytes = new Base64().encode(FileUtils.readFileToByteArray(image));
+    } catch (IOException e) {
+      log("logImage", e.getMessage(), false);
+    }
+    image(command, new String(bytes, StandardCharsets.UTF_8), success);
+  }
+
+  public static void image(String command, String imageAsBase64, boolean success) {
+    String imgHtml = VelocityWrapper.fillImage(imageAsBase64);
+    VelocityWrapper
+        .fillLogRow(Arrays.asList(success ? LogLevel.OK : LogLevel.ERROR), command, imgHtml);
+  }
+
+  public static void logJSError() {
+    if ("true".equals(Configuration.getJSErrorsEnabled())) {
+      JavascriptExecutor js = DriverProvider.getActiveDriver();
+      List<String> error =
+          (ArrayList<String>) js.executeScript("return window.JSErrorCollector_errors.pump()");
+      if (!error.isEmpty()) {
+        VelocityWrapper.fillLogRow(Arrays.asList(LogLevel.ERROR), "click", error.toString());
+      }
+    }
+  }
+
+  public static List<Boolean> getVerificationStack() {
+    return LOGS_RESULTS;
+  }
+
+  public static void startTest(Method testMethod) {
+    String testName = testMethod.getName();
+    String className = testMethod.getDeclaringClass().getCanonicalName();
+    String command;
+    String description;
+    if (testMethod.isAnnotationPresent(RelatedIssue.class)) {
+      String issueID = testMethod.getAnnotation(RelatedIssue.class).issueID();
+      String jiraPath = "https://wikia-inc.atlassian.net/browse/";
+      String jiraUrl = jiraPath + issueID;
+      String jiraLink = VelocityWrapper.fillLink(jiraUrl, issueID);
+      command = "Known failure";
+      description =
+          testName + " - " + jiraLink + " " + testMethod.getAnnotation(RelatedIssue.class)
+              .comment();
+    } else {
+      command = "";
+      description = testName;
+    }
+    String html = VelocityWrapper.fillFirstLogRow(className, testName, command, description);
+    CommonUtils.appendTextToFile(LOG_PATH, html);
+    testStarted = true;
+  }
+
+  public static void logAssertionStacktrace(AssertionError exception) {
+    WikiaWebDriver driver = DriverProvider.getActiveDriver();
+
+    Log.imageCounter += 1;
+    if ("true".equals(Configuration.getLogEnabled())) {
+      String exceptionMessage = ExceptionUtils.getStackTrace(exception);
+      List<LogData> classList = new ArrayList<>();
+      classList.add(LogLevel.ERROR);
+      classList.add(LogType.STACKTRACE);
+      String html = VelocityWrapper.fillErrorLogRow(classList, exceptionMessage, Log.imageCounter);
+      try {
+        new Shooter().savePageScreenshot(Log.SCREEN_PATH + Log.imageCounter, driver);
+        CommonUtils.appendTextToFile(Log.SCREEN_PATH + Log.imageCounter + ".html",
+                                     Log.getPageSource(driver));
+        CommonUtils.appendTextToFile(Log.LOG_PATH, html);
+      } catch (Exception e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)